app/components/local_time_component.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class LocalTimeComponent < ViewComponent::Base
+  include ActionController::Cookies
+  attr_reader :format, :unix_timestamp
+
+  def initialize(format:, unix_timestamp:)
+    @format = format
+    @unix_timestamp = unix_timestamp
+  end
+
+  def local_time
+    # Time format should be passed as 12 or 24 only
+    unless [12, 24].include?(format)
+      raise ArgumentError, "Invalid time format argument"
+    end
+
+    time = Time.at(unix_timestamp).in_time_zone(@time_zone)
+    time_format = format == 12 ? "%I:%M %p" : "%H:%M"
+    formatted_date = time.strftime("%B %d, %Y")
+    time_of_day = time.strftime(time_format)
+    time_zone = time.zone
+    formatted_date + " at " + time_of_day + " " + time_zone
+  end
+
+  private
+
+  def before_render
+    browser_tz = ActiveSupport::TimeZone.find_tzinfo(cookies[:browser_time_zone])
+    ActiveSupport::TimeZone.all.find { |zone| zone.tzinfo == browser_tz } || Time.zone
+  rescue TZInfo::UnknownTimezone, TZInfo::InvalidTimezoneIdentifier
+    @time_zone = Time.zone
```

Comment:
we need this from the client. Can you add this as a parameter for the component?

Reply:
Hi @FireLemons, this is the local time zone of the user taken from browser. The arguments required for the component and mentioned in the issue are the time in unix timestamp format, and the format. So do I need to pass the time zone as the third argument?

Reply:
Yes, it's from the client